syntax = "proto3";

package user;


service UserService {
  rpc create_room(CreateRoom) returns (CreateRoomResponse) {}

  rpc join_room(JoinRoom) returns (JoinRoomResponse) {}

  //  rpc get_message(Request) returns (Response) {}
  //
  //  rpc send_message(User) returns (Token) {}
  //
}

message User {
  int32 room_code = 1;
  string name = 2;
}

message Request {}

message Response {
  User user = 1;
  repeated User users = 2; //array of users
  repeated Error errors = 3; //array of errors if such happened
}

message Token {
  string token = 1;
  bool valid = 2;
  repeated Error errors = 3; //array of errors if such happened
}

message Error {
  int32 code = 1;
  string description = 2;
}

message JoinRoom{
  string room_name = 1;
}

message JoinRoomResponse{
  Settings settings = 1;
}

message CreateRoom{
  //  string name = 1;
  Settings settings = 1;
}

message CreateRoomResponse{
  int32 id = 1;
}

message Settings{
  string room_name = 1;
  int32 quantity_of_players = 2;
  int32 size_of_deck = 3;
  int32 time_of_move = 4;
  int32 local_player = 5;
}